coordinator/src/trade/mod.rs

```diff
@@ -520,7 +563,179 @@ impl TradeExecutor {
+            collateral_reserve_trader,
+            realized_pnl,
+        } = resized_position;
+
+        let contract_descriptor = payout_curve::build_contract_descriptor(
+            average_execution_price,
+            margin_coordinator.to_sat(),
+            margin_trader.to_sat(),
+            leverage_coordinator,
+            leverage_trader,
+            coordinator_direction,
+            collateral_reserve_coordinator.to_sat(),
+            collateral_reserve_trader.to_sat(),
+            contracts.to_f32().expect("to fit"),
+            trade_params.contract_symbol,
+        )
+        .context("Could not build contract descriptor")?;
+
+        let contract_symbol = trade_params.contract_symbol.label();
+        let expiry_timestamp = trade_params.filled_with.expiry_timestamp;
+        let expiry_unix_timestamp = expiry_timestamp.unix_timestamp();
+
+        let sats_per_vbyte = self
+            .node
+            .inner
+            .fee_rate_estimator
+            .get(ConfirmationTarget::Normal)
+            .as_sat_per_vb()
+            .round();
+        // This fee rate is actually ignored since the fee reserve is defined when the channel is
+        // first opened.
+        let fee_rate = Decimal::try_from(sats_per_vbyte)?
+            .to_u64()
+            .context("failed to convert to u64")?;
+
+        // The contract input to be used for setting up the trade between the trader and the
+        // coordinator.
+        let event_id = format!("{contract_symbol}{expiry_unix_timestamp}");
+
+        tracing::debug!(
+            event_id,
+            oracle=%trade_params.filled_with.oracle_pk,
+            "Proposing DLC channel update"
+        );
+
+        let contract_input = ContractInput {
+            offer_collateral: margin_coordinator.to_sat() + collateral_reserve_coordinator.to_sat(),
+            accept_collateral: margin_trader.to_sat() + collateral_reserve_trader.to_sat(),
+            fee_rate,
+            contract_infos: vec![ContractInputInfo {
+                contract_descriptor,
+                oracles: OracleInput {
+                    public_keys: vec![to_xonly_pk_29(trade_params.filled_with.oracle_pk)],
+                    event_id,
+                    threshold: 1,
+                },
+            }],
+        };
+
+        let protocol_id = ProtocolId::new();
+        let channel = self.node.inner.get_dlc_channel_by_id(&dlc_channel_id)?;
+        let previous_id = match channel.get_reference_id() {
+            Some(reference_id) => Some(ProtocolId::try_from(reference_id)?),
+            None => None,
+        };
+
+        let temporary_contract_id = self
+            .node
+            .inner
+            .propose_dlc_channel_update(&dlc_channel_id, contract_input, protocol_id.into())
+            .await
+            .context("Could not propose DLC channel update")?;
+
+        // TODO: Shouldn't we include actually starting the DLC protocol in the DB transaction in
```

Comment:
Since we use different transactional systems, postgres and sled I think it doesn't matter. But once we are also using postgres that would be great. As we could atomically execute both rust dlc and 10101 changes.